code-challenges.test.js

```diff
@@ -14,19 +14,93 @@
+
+ // I am going to first start with the expect statements, one for each variable 
+ // Since I am not familiar with the Fibonacci sequence I will have to do some research 
+ // In the describe arguement I will put fibonacci 
+// The next step is where I felt I was having the most difficulty, the layout of my expect statement. I was making it more complicated than needed. This time I will put fibonacci to invoke the function and the expected outputs on the right to be compared to 
+
 const fibLength1 = 6
 // Expected output: [1, 1, 2, 3, 5, 8]
 
 const fibLength2 = 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+// describe("fibonacci", () => { 
+//   it("takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence", () => {
+//     expect(fibonacci(fibLength1)).toEqual([1, 1, 2, 3, 5, 8])
+//     expect(fibonacci(fibLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+//   }) 
+// })
+
+// I ran the test and at first I didn't get the correct error, I "got Missing second argument. It must be a callback function. Perhaps you want to use `test.todo` for a test placeholder." So I thought maybe syntax error with the parentheses. I fixed the parentheses and then I did get "ReferenceError: fibonacci is not defined"
+
 // b) Create the function that makes the test pass.
 
+//Now for the fibonacci function
+//Input:  fibLength1
+//Output: [1, 1, 2, 3, 5, 8]
+//Input: fibLength2
+//Output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+// Looking into the Fibonacci sequence it works by finding the next number by adding the last 2 numbers before it. 
+//I found this concept called recursion that is basically a for loop as it uses the same rules. Its a function that calls upon itself until a condition is met. 
+
+// function fibonacci(num) 
+//    {    
+//         if(num==1) 
+//             return 0; 
+//         if (num == 2) 
+//             return 1; 
+//         return  fibonacci(num - 1) + fibonacci(num - 2);
+//    } 
+//This didn't give me what I wanted, this would only return the number but not a list of numbers. 
+
+//I found a while loop which will loop while a condition is true and will stop when the condition is false, so if I ask it to give me 6 numbers it will keep going until it reaches 6 then it will stop.
+
+// var fibonacci = function (n) 
+// {
+//   if (n===1) 
+//   {
+//     return [0, 1];
+//   } 
+//   else 
+//   {
+//     var s = fibonacci(n - 1);
+//     s.push(s[s.length - 1] + s[s.length - 2]);
+//     return s;
+//   }
+// }
+
+// I get this error, this is as close as I can get  
+    // - Expected  - 0
+    // + Received  + 1
+
+    // @@ -1,6 +1,7 @@
+    //   Array [
+    // +   0,
+    //     1,
+    //     1,
+    //     2,
+    //     3,
+    //     5,
+// I have tried again for too long, for 3 hours trying different changes trying to get it to start one index higher but I cannot get it to work no matter what.
```

Comment:
You were heading in the right direction with the idea of using a while loop.  A for loop is one of the most common ways to write out this problem since we are given a number that is determining the length of the array we will get back.  We can then add to an array the last two iterations added together to be the current iteration.